jgiven-tests/src/test/java/com/tngtech/jgiven/impl/GuaranteedStateTest.java

```diff
@@ -0,0 +1,110 @@
+package com.tngtech.jgiven.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.tngtech.jgiven.Stage;
+import com.tngtech.jgiven.annotation.BeforeStage;
+import com.tngtech.jgiven.annotation.ProvidedScenarioState;
+import com.tngtech.jgiven.exception.JGivenMissingGuaranteedScenarioStateException;
+import com.tngtech.jgiven.junit.ScenarioTest;
+import com.tngtech.jgiven.junit.SimpleScenarioTest;
+import com.tngtech.jgiven.report.model.ReportModel;
+import com.tngtech.jgiven.testframework.TestExecutor;
+import com.tngtech.jgiven.testframework.TestFramework;
+import com.tngtech.jgiven.tests.TestScenarioRepository;
+import org.junit.Test;
+
+public class GuaranteedStateTest extends SimpleScenarioTest<GuaranteedStateTest.SimpleTestStage> {
+
+    @Test
+    public void assure_before_method_of_second_test_is_executed_after_guaranteed_fields_validation() {
+        given().a_Jgiven_test_with_a_guaranteed_null_state();
+        when().the_test_is_executed();
+        then().the_report_contains_$_exception(JGivenMissingGuaranteedScenarioStateException.class);
+    }
+
+    @Test
+    public void assure_before_method_of_second_test_is_executed_if_guaranteed_initialized() {
+        given().a_Jgiven_test_with_a_guaranteed_state();
+        when().the_test_is_executed();
+        then().the_report_contains_$_exception(ClassNotFoundException.class);
+    }
+
+    public static class RealTest extends ScenarioTest<RealTest.RealGiven, RealTest.RealWhen, RealTest.RealThen> {
+
+        @Test
+        @SuppressWarnings("checkstyle:MethodName")
```

Comment:
I think a better approach to solving this is to check whether we can disable some checkstyle rules for test classes.